feat(promtail): add more complex options of dropping logs at the drop stage (#6371)
grafana/loki: CHANGELOG.md

##### Fixes
* [6034](https://github.com/grafana/loki/pull/6034) **DylanGuedes**: Promtail: Fix symlink tailing behavior.
##### Changes

> * [6371](https://github.com/grafana/loki/pull/6371) **witalisoft**: Drop stage with a complex match based on multiple extracted data fields, breaking config changes regarding to drop stage section

Reviewer: I think it's enough to indicate the breaking change and omit details about it in the changelog, because the upgrade docs contain that information:

```suggestion
* [6371](https://github.com/grafana/loki/pull/6371) **witalisoft**: BREAKING: Support more complex match based on multiple extracted data fields in drop stage
```
